fvm/src/syscalls/mod.rs

```diff
@@ -29,17 +30,56 @@ pub struct InvocationData<K> {
     /// The last-seen syscall error. This error is considered the abort "cause" if an actor aborts
     /// after receiving this error without calling any other syscalls.
     pub last_error: Option<backtrace::Cause>,
+
+    /// A snapshot is _taken_ every time execution returns _from_ WASM.
+    /// This happens at the start of a syscall invocation as well as the end of WASM invocation.
+    /// The snapshot is _updated_ every time execution moves _into_ WASM.
+    /// This happens at the first WASM invocation, and at the end of a syscall invocation.
+    /// Before a snapshot is _updated_, fuel is consumed based on the delta.
+    pub gas_available_snapshot: i64,
+
+    /// A snapshot is _taken_ every time execution moves _into_ WASM.
```

Comment:
These comments are mildly confusing. After reading them it's not clear to me what the difference between taking a snapshot and updating a snapshot is?

Reply:
Clarifying, thanks for the feedback.